DOCS-3038: CORE Encryption Article Changes (#1244)
truenas/documentation: content/CORE/Storage/Pools/StorageEncryption.md

The local TrueNAS system manages keys for data-at-rest.
The user is responsible for storing and securing their keys.

> TrueNAS, as of 12., includes the [Key Management Interface Protocol (KMIP)](https://docs.oasis-open.org/kmip/spec/v1.1/os/kmip-spec-v1.1-os.html).

Reviewer: Quick suggestion to smooth out the wording and fix the version reference.
```suggestion
TrueNAS 12.0 and newer includes the [Key Management Interface Protocol (KMIP)](https://docs.oasis-open.org/kmip/spec/v1.1/os/kmip-spec-v1.1-os.html).
```

![StoragePoolsDatasetUnlockOptions](/images/CORE/12.0/StoragePoolsDatasetUnlockOptions.png "Dataset Unlock Options")

Type the passphrase and click **SUBMIT**. 
To unlock child datasets, set the **Unlock Children** checkbox. 
Child datasets that inherited the parent dataset's encryption settings unlock when the parent unlocks. 
Users can unlock child datasets with different passphrases as the parent simultaneously by entering their passphrases.

Two confirmation windows display. 
Click **CONTINUE** to confirm you want to unlock the datasets or **CANCEL** to exit and keep the datasets locked. 
A second confirmation window displays confirming the datasets are unlocked. 
Click **CONTINUE** to close the second confirmation window. 

![StoragePoolsDatasetUnlockSuccess](/images/CORE/12.0/StoragePoolsDatasetUnlockSuccess.png "Dataset Unlock Success")

TrueNAS displays the dataset with the unlocked icon.

**Unlocking Mutliple Datasets Example:**

![StoragePoolsDatasetUnlockexample1](/images/CORE/12.0/EncrytionExample1.png "Encrypted locked Datasets")

> The parent dataset is media. 
> It has three child datasets. 
> The `documents` child datasetinherits the parent encryption settings and its password. 
> The other two child datasets (`audio` and `video`) have their own passphrases. 

Reviewer: Suggested syntax switch to embolden names of datasets called out from example image:
```suggestion
The parent dataset is **media**.
It has three child datasets.
The **documents** child dataset inherits the parent encryption settings and its password.
The other two child datasets (**audio** and **video**) have their own passphrases.
```

After locking the parent dataset all child datasets lock too.

![StoragePoolsDatasetUnlockexample2](/images/CORE/12.0/EncrytionExample2.png "Password for locked Datasets")

Open the <i class="material-icons" aria-hidden="true" title="Options">more_vert</i> for the parent dataset and click **Unlock**. 
To unlock all the datasets, click on the **Unlock Children** checkbox to check-mark it and type the passphrase for each dataset to unlock them.

![StoragePoolsDatasetUnlockexample3](/images/CORE/12.0/EncrytionExample3.png "Successfully unlocked Datasets")

Click the **CONTINUE** button on both confirmation windows to successfully unlock the datasets. 

> The datasets display the unlocked icon.
> ## Encrypting a Zvol

Reviewer: Note particularly necessary for the final rendered article but suggested an empty line at the end of the section as a readability improvement for those looking at the raw text.
```suggestion
The datasets display the unlocked icon.

## Encrypting a Zvol
```

Encryption is for securing sensitive data. 

{{< hint info >}}

> You can only encrypting a zvol if you create the zvol from a dataset with encryption.

Reviewer: grammo
```suggestion
You can only encrypt a zvol if you create the zvol from a dataset with encryption.
```

{{< /hint >}}

{{< include file="/_includes/EncryptionBackupKeys.md" type="page" >}}

Zvols, like datasets, inherit encryption settings from the parent dataset. 
To encrypt a zvol, select a dataset configured with encryption and then [create a new zvol]({{< relref "/CORE/Storage/Pools/zvols.md" >}}).
Next, click the <i class="material-icons" aria-hidden="true" title="Options">more_vert</i> icon to display the **Zvol Actions** menu. 

![AddZvolActionsMenuWithEncryptionOptions](/images/CORE/12.0/AddZvolActionsMenuWithEncryptionOptions.png "Zvol Actions with Encryption Options")

> If you do not see encryption options on the menu you created the zvol from a dataset not configured with encryption. You can delete the zvol and start over.

Reviewer: Not sure here, feels like it needs a comma or **then**:
```suggestion
If you do not see encryption options on the menu then you created the zvol from a dataset not configured with encryption. You can delete the zvol and start over.
```

![StoragePoolsEncryptionActionsExportKeys](/images/CORE/12.0/StoragePoolsEncryptionActionsExportKeys.png "Exporting Key Files")

{{< hint info >}}
To manually download a backup a single dataset's key file, click the dataset <i class="material-icons" aria-hidden="true" title="Options">more_vert</i> and click on **Export Key**. 
Next type the root password and click the **CONTINUE** button. Click the **DOWNLOAD KEY** button.
{{< /hint >}}

To change the key, click the dataset <i class="material-icons" aria-hidden="true" title="Options">more_vert</i> and **Encryption Options**.

![StoragePoolsEncryptedDataset](/images/CORE/12.0/StoragePoolsEncryptedDataset.png "Dataset Options: Encryption Options")

Type your custom key or click **Generate Key**.

![StoragePoolsEncryptedDatasetOptions](/images/CORE/12.0/StoragePoolsEncryptedDatasetOptions.png "Editing Encryption Options")

{{< /tab >}}
{{< tab "Passphrases" >}}

> To use a passphrase instead of a keyfile, click the dataset <i class="material-icons" aria-hidden="true" title="Options">more_vert</i> icon to dispaly the **Dataset Actions** menu and then click **Encryption Options**.

Reviewer: typo
```suggestion
To use a passphrase instead of a keyfile, click the dataset <i class="material-icons" aria-hidden="true" title="Options">more_vert</i> icon to display the **Dataset Actions** menu and then click **Encryption Options**.
```

### GELI Pool Migrations

You can *migrate* data  from the GELI-encrypted pool to a new ZFS-encrypted pool.

{{< hint warning >}}
Be sure to unlock the GELI-encrypted pool before attempting any data migrations.
The new ZFS-encrypted pool must be at least the same size as the previous GELI-encrypted pool.
Do not delete the GELI dataset until you have verified the data migration.
{{< /hint >}}

There are a few options to migrate data from a GELI-encrypted pool to a new ZFS-encrypted pool:

{{< tabs "GELI Migration Methods" >}}
{{< tab "Replication Wizard" >}}
{{< expand "Using the Replication Wizard" "v" >}}

> TrueNAS web interface continues to detect and suppot GELI encrypted pools as *Legacy Encrypted* pools. As of TrueNAS version 12.0-U1, a decrypted GELI pool can migrate data to a new ZFS encrypted pool using the Replication Wizard.

Reviewer: Grammo & typo
```suggestion
The TrueNAS web interface continues to detect and support GELI encrypted pools as *Legacy Encrypted* pools. As of TrueNAS version 12.0-U1, a decrypted GELI pool can migrate data to a new ZFS encrypted pool using the Replication Wizard.
```

 
Start the Replication Wizard by selecting **Tasks** -> **Replication Task** -> **ADD**

**Source Location**:
 * Select **On this System**.
 * Set the dataset to transfer.
 
**Destination Location**:
 * Select **On a Different System**.
 
**SSH Connection**:
 * Either create the ssh connection by clicking **Create New** or select the destination system's ssh connection.
 * In **Destination**, select the dataset to replicate the files to.
 * Set **Encryption**.
 * Choose either **PASSPHRASE** or **HEX** for the **Encryption Key Format**.

>  * If you selected PASSPHRASE, type the passphrase. If you selected HEX, set *G*enerate Encryption Key**.

Reviewer: syntax
```suggestion
 * If you selected PASSPHRASE, type the passphrase. If you selected HEX, set **Generate Encryption Key**.
```
